src/Promise.php

```diff
@@ -7,4 +7,5 @@
  */
 interface Promise
 {
+    public function cancel();
```

Comment:
```suggestion
    /**
    * Requests the cancellation of the promise.
    * Functions still yielding for this promise will receive a `CancelledException`.
    */
    public function cancel(string $reason): void;
```
* Since php 7.1, use `void` when nothing is returned.
* What about the `$reason` parameter? To be forwarded in the exception (and documented)